proxy/envoy/config.go

```diff
@@ -402,3 +516,7 @@ func buildFilters(mesh *MeshConfig) []Filter {
 
 	return filters
 }
+
+func clusterIdentifierString(identifier *config.ClusterIdentifier) string {
```

Comment:
This is leaking service key encoding scheme. Can you create a service with hostname = identifier.name, and tags = identifier.Tags and then call String().